rust/parser/src/lexer.rs

```diff
@@ -960,40 +980,42 @@ impl<'alloc> Lexer<'alloc> {
                         }
                     }
                 }
+
                 'u' => {
                     let c = self.unicode_escape_sequence_after_backslash_and_u()?;
                     text.push(c);
                 }
+
                 '0' => {
                     // In strict mode code and in template literals, the
                     // relevant production is
                     //
                     //     EscapeSequence ::
                     //         `0` [lookahead <! DecimalDigit]
                     //
-                    // In non-strict StringLiterals, legacy octal escape
-
-                    // LegacyOctalEscapeSequence ::
-                    //     OctalDigit [lookahead <! OctalDigit]
-                    //     ZeroToThree OctalDigit [lookahead <! OctalDigit]
-                    //     FourToSeven OctalDigit
-                    //     ZeroToThree OctalDigit OctalDigit
-                    //
-                    // ZeroToThree :: one of
-                    //     `0` `1` `2` `3`
-                    //
-                    // FourToSeven :: one of
-                    //     `4` `5` `6` `7`
-
-                    if let Some('0'..='9') = self.peek() {
-                        return Err(ParseError::InvalidEscapeSequence);
+                    // In non-strict StringLiterals, `\0` begins a
+                    // LegacyOctalEscapeSequence which may contain more digits.
+                    match self.peek() {
+                        Some('0'..='7') => {
+                            return Err(ParseError::NotImplemented(
+                                "legacy octal escape sequence in string",
+                            ));
+                        }
+                        _ => {}
```

Comment:
might be nice to mention about the difference between the spec and implementations about `\08` and `\09` here,
in the same way as `\8` and `\9` below